apis/v1alpha1/clusterprofile_types.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ClusterProfileSpec defines the desired state of ClusterProfile
+type ClusterProfileSpec struct {
+	// DisplayName defines a human-readable name of the ClusterProfile
+	DisplayName string `json:"displayName"`
+
+	// ClusterManager defines which cluster manager created this ClusterProfile resource
+	ClusterManager ClusterManager `json:"clusterManager"`
+}
+
+// ClusterManager defines which cluster manager created this ClusterProfile resource.
+// It's recommended that each cluster manager instance should set a different values to this field.
+// In addition, it's recommended that a predefined label with key "x-k8s.io/cluster-manager"
+// should be added by the cluster manager upon creation.
+// The value of the label should be the same as the name of the cluster manager.
+// The purpose of this label is to make filter clusters from different cluster managers easier.
+type ClusterManager struct {
+	// Name defines the name of the cluster manager
+	Name string `json:"name"`
+}
+
+// ClusterProfileStatus defines the observed state of ClusterProfile
+type ClusterProfileStatus struct {
+	// Conditions contains the different condition statuses for this cluster.
+	Conditions []metav1.Condition `json:"conditions"`
+
+	// Version defines the Kubernetes version of the cluster.
```

Comment:
```suggestion
	// Version defines the version information of the cluster.
```
we may incorporate other version, eg, enabled featureset version

Reply:
Done as suggested. Thanks.